### PR TITLE
Remote Config Testing Console  API

### PR DESCRIPTION
**Pull request by ncooke3**
Adding a testing API (and some tests to showcase it) for interacting with the Remote Config console. Included a README with instructions on how to run the added tests and play with the testing API. With this addition, we could run tests to test remote config's implementation using the fake console API when we don't need to test with a live console, and this new api when we do want to test with a console.

### CODE REVIEW

File: FirebaseRemoteConfig/Tests/SwiftAPI/APITests.swift [hunk 1]
```diff
@@ -18,10 +18,24 @@ import FirebaseCore
 import XCTest
 
 class APITests: APITestBase {
+  var console: RemoteConfigConsole!
+
   override func setUp() {
     super.setUp()
     if APITests.useFakeConfig {
       fakeConsole.config = ["Key1": "Value1"]
+    } else {
```
Comment (paulb777) [inline FirebaseRemoteConfig/Tests/SwiftAPI/APITests.swift:27]: Shouldn't there be three cases here?  1. fakeConsole 2. RemoteConfigConsole() 3. Real console.

Comment (ncooke3) [reply FirebaseRemoteConfig/Tests/SwiftAPI/APITests.swift:27]: Yes, you're right!

**Comment by ncooke3** [comment]
Going to open a new PR for this because i messed this one up 🤦‍♂️

**Comment by ncooke3** [comment]
Reopening for fix

**Comment by ncooke3** [comment]
Closing in favor of updated PR

**Pull request closed without merging**